src/Vendr.DemoStore/Web/Controllers/SearchSurfaceController.cs

```diff
@@ -30,33 +29,35 @@ public SearchSurfaceController(IExamineManager examineManager)
         }
 
         [ChildActionOnly]
-        public ActionResult Search(string q = "", int p = 1, int ps = 12)
+        public ActionResult Search([FromUri] SearchQuery request)
         {
             // The logic for searching is mostly pulled from ezSearch
             // https://github.com/umco/umbraco-ezsearch/blob/master/Src/Our.Umbraco.ezSearch/Web/UI/Views/MacroPartials/ezSearch.cshtml
 
             // Prepair a view model to return
-            var result = new SearchViewModel();
+            var model = new SearchViewModel();
+
+            // Check if the searcher is available
+            if (_examineManager.TryGetSearcher("FacetSearcher", out ISearcher searcher) == false)
+            {
+                throw new Exception("Failed to find searcher");
+            }
 
-            // Populate category names collection which will be used to
-            // provide friendly names for the facet categories
-            var homePage = CurrentPage.GetHomePage();
-            var categoriesNode = homePage.Children.OfType<CategoriesPage>().FirstOrDefault();
-            var categoryNodes = categoriesNode.Children.OfType<CategoryPage>();
-            result.CategoryNames = categoryNodes.ToDictionary(x => x.UrlSegment.MakeSearchTermSafe(), x => x.Name);
+            // Build the query
+            var query = searcher.CreateQuery("Content")
```

Comment:
@callumbwhyte maybe use the `IndexTypes.Content` constant here in `Umbraco.Examine` namespace?

Reply:
Yes, I was trying to be quick on my stream and never came back to this... Will fix!